### DISCUSSION TITLE
GLTF viewer, emscripten and build_mode

### DISCUSSION THREAD

**Discussion by JD31** [DiligentGraphics/DiligentSamples#98] [Q&A]
Hello !

(I am very beginner in the world of web assembly and emscripten compiler! )

I am able to run the GLTFViewer into chrome when I build in Debug mode with Emscripten.
But, if building exactly the same source source in Release mode (or RelWithDebInfo), it doesn't work.
There is an exception with the following stack:

> GLTFViewer.html:1 Uncaught RuntimeError: memory access out of bounds
>     at 00b0e1c6:0xdd4ec
>     at 00b0e1c6:0xe6edf
>     at 00b0e1c6:0x17a5af
>     at 00b0e1c6:0x1944a6
>     at 00b0e1c6:0x191f11
>     at Module._main (GLTFViewer.html:1:3985291)
>     at callMain (GLTFViewer.html:1:3991097)
>     at I (GLTFViewer.html:1:3991361)
>     at GLTFViewer.html:1:3991571

Any ideas ?

**Comment by TheMostDiligent** [comment]
Did you try to run other applications? What about Tutorial01 and Tutorial03 in particular?

**Reply by JD31** [reply]
Yep. Both of them works well on Debug and Release mode.

**Comment by TheMostDiligent** [comment]
Did you try to run it in another browser or platform? It is little bit hard to tell, could be browser bug or emscripten bug. Could also be a bug in the GLTF viewer, but it works fine on all other platforms.

**Reply by JD31** [reply]
Idem with Firefox, MS Edge and Safari.

**Reply by JD31** [reply]
Work well on native with MSVC in all build_modes

**Reply by TheMostDiligent** [reply]
Maybe one thing to try is to play with compiler settings. I have seen high levels of optimization  producing invalid executables.
